deploy/aws/java11Exec/src/main/java/poseidon/App.java

```diff
@@ -120,6 +122,53 @@ private File writeFS(Map<String, String> files) throws IOException {
         return workspace;
     }
 
+    private String[] prepareCommand(AwsFunctionRequest execution) {
+        String[] cmd = execution.cmd;
+        String innerCommand = cmd[cmd.length - 1]; // We expect a "sh -c" wrapped command.
+        String replacedInnerCommand = simpleMakefileReplacement(innerCommand, execution.files);
+        cmd[cmd.length - 1] = replacedInnerCommand;
+        return cmd;
+    }
+
+    protected String simpleMakefileReplacement(String command, Map<String, String> files) {
+        Pattern isMakeCommand = Pattern.compile("^make\\s?(\\w*)?$");
```

Comment:
We might want to introduce a non-capturing group and (maybe) a named capturing group: `^make(?:\s+(?'name'\w*)?)$#`

Reply:
Nice regex feature and fancy that you know about them.
How about `^make(?:\s+(?<startRule>\w*))?$` and `(?<name>.*):\n(?<commands>(?:\t.+\n?)*)`

Reply:
Yes, that looks fine!